test/regex_test.cpp

```diff
@@ -0,0 +1,186 @@
+#include "gtest/gtest.h"
+#include "regex_yaml.h"
+#include "stream.h"
+
+using YAML::RegEx;
+using YAML::Stream;
+
+namespace {
+TEST(RegExTest, Empty) {
+  RegEx empty;
+  EXPECT_TRUE(empty.Matches(std::string()));
+  EXPECT_EQ(0, empty.Match(std::string()));
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
+    EXPECT_FALSE(empty.Matches(char(ii)));
+    EXPECT_EQ(-1, empty.Match(char(ii)));
+  }
+}
+
+TEST(RegExTest, SingleCharacter) {
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
+    RegEx ex((char)ii);
+    for (int jj = Stream::eof() + 1; jj < 128; ++jj) {
+      if (jj == ii) {
+        EXPECT_TRUE(ex.Matches(char(jj)));
+        EXPECT_EQ(1, ex.Match(char(jj)));
+
+        // Match at start of string only!
+        std::string prefixed = std::string(1, ii + 1) +
+                               std::string("prefix: ") +
+                               std::string(1, char(jj));
+        EXPECT_FALSE(ex.Matches(prefixed));
+        EXPECT_EQ(-1, ex.Match(prefixed));
+      } else {
+        EXPECT_FALSE(ex.Matches(char(jj)));
+        EXPECT_EQ(-1, ex.Match(char(jj)));
+      }
+    }
+  }
+}
+
+TEST(RegExTest, Range) {
+  for (int ii = Stream::eof() + 1; ii < 127; ++ii) {
```

Comment:
I intended to test j > i as well. I figured testing everything is easy and cheap.